types/src/transaction/script.rs

```diff
@@ -50,3 +51,80 @@ impl fmt::Debug for Script {
             .finish()
     }
 }
+
+/// How to call a particular Move script (aka. an "ABI").
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub struct ScriptABI {
+    /// The public name of the script.
+    name: String,
+    /// Some text comment.
+    doc: String,
+    /// The `code` value to set in the `Script` object.
+    code: Vec<u8>,
+    /// The names of the type arguments.
+    ty_args: Vec<String>,
+    /// The description of regular arguments.
+    args: Vec<ArgumentABI>,
```

Comment:
`params` instead of `args`?

Reply:
I agree but this ship has sailed: the other definitions use `args` consistently. I think I even saw a TODO to remove the last `param`

Reply:
Well still I think the right thing to do is to replace all other `args` with `params`. These two terms have clear meanings and are not really interchangeable in the PL context... with that being said I totally understand this is not a high pri so please don't feel obligated to correct it in this PR.

(To clarify, by "other definitions" you mean things outside `language` right? AFAIK all except 1 or 2 definitions in language use the correct term.)

Reply:
Other definitions in `types/src/transactions`.

Reply:
fwiw, @vgao1996 is referring to what the binary format uses and what we have tried to move the code base to. And by codebase we mean the one in language. 
So in that respect, if the intent was consistency with the `language` part of the code `parameters` is what we went for. 
Of course even parts within `language` are not consistent yet